sound/soc/sof/intel/hda.c

```diff
@@ -1359,8 +1272,24 @@ struct snd_soc_acpi_mach *hda_machine_select(struct snd_sof_dev *sdev)
 		/* report to machine driver if any DMICs are found */
 		mach->mach_params.dmic_num = check_dmic_num(sdev);
 
+		/* SDW mach does not use DMIC quirk flag. */
```

Comment:
I can't figure out what this comment means, and same for the commit message

SDW mach does not use the DMIC quirk flag and pass full topology name
with file extension to SOF driver. Therefore, we add extra code to
hda_generic_machine_select() function to cope with the difference.

if the goal was to unify, why do we still need differences with SoundWire handling for the dmic case? it's the same in the end, we need the -2ch or -4ch suffix appended, no?

Reply:
Sorry my wording is misleading. In this PR I am trying to use same code to do topology name fixup for all I2S/HDA/SDW mach. There is no functional change in these commits. I'm updating the git commit/comment for more accurate description.